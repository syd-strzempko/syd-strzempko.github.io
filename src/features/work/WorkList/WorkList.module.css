.title {
    text-align: center;
    font-size: 1.25rem;
    padding-bottom: 10px;
}
.wrapper {
    display: flex;
    flex-flow: row nowrap;
    width: 100%;
}
.line {
    border-left: 3px solid var(--link-active);
    padding-left: 11px;
    margin: 20px 0;
}
.listWrapper {
    display: flex;
    flex-flow: column nowrap;
}
.nonlined {
    padding-left: 13.5px;
}
.listItem{
    display: flex;
    align-items: center;
    gap: 10px;
    padding: 5px 0;
}
.listRole {
    color: var(--beige1);
    font-style: italic;
}
.point {
    content: '';
    width: 11px;
    height: 11px;
    border-radius: 50%;
    border: 3px solid var(--link-active);
    background-color: var(--white);
    margin-left: -20px;
}
.point:hover {
    background-color: var(--link-active);
}
.tooltip {
    background-color: var(--white) !important;
    border: 2px solid var(--light2) !important;
    border-radius: 15px !important;
    color: var(--beige2) !important;
    opacity: 1 !important;
}