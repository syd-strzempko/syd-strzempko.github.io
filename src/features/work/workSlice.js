import { createSlice } from '@reduxjs/toolkit';

export const slice = createSlice({
  name: 'work',
  initialState: {
    jobs: [
        {
            url: 'sunpower',
            org: 'SunPower Corp.',
            title: 'Full Stack Software Engineer',
            project: null,
            link: 'https://us.sunpower.com/products/software/mysunpower',
            span: '2022-2024',
            details: [
                'Built out new feature API and UI framework, communicated with PM and cross-team developers to establish new data entity expectations',
                'Integrated AWS Chat Client into app to improve and track customer intermediary feedback/troubleshooting',
                'Leveraged Google Analytics GA4 custom dimensions and created explorations to garner insight into customer activity'
            ]
        },
        {
            url: 'fulcrum',
            org: 'Fulcrumpro Inc.',
            title: 'Software Engineer',
            project: null,
            link: 'https://fulcrumpro.com/',
            span: '2020-2021',
            details: [
                'Aggregated, transformed, and projected RxJS observables from numerous endpoints into component-specific response streams',
                'Implemented asynchronous DB LINQ queries on endpoints to efficiently fetch multiple data sources for response',
                'Integrated PM feedback to create polished, flexible UI components'
            ]
        },
        {
            url: 'bertha',
            org: 'Aunt Bertha',
            title: 'Full Stack Developer',
            project: null,
            link: 'https://www.findhelp.org/',
            span: '2020-2020',
            details: [
                'Utilized custom time-bounded digest decorator on API endpoint to aggregate requests into a standardized response behavior',
                'Architecture reviewed, gathered feedback on, and implemented new relational DB architecture to expand business logic insight',
                'Debugged crucial email-embedded CSRF token workflow for single-use update links provided for clients'
            ]
        },
        {
            url: 'yonder',
            org: 'Yonder',
            title: 'Full Stack Developer',
            project: null,
            link: 'https://www.yonder-ai.com/',
            span: '2019-2020',
            details: [
                'Wrote and standardized CRUD API operations across services',
                'Wrote complex SQL queries and upsertions to aggregate analytics data and move their output between databases',
                'Performed Alembic database migrations to modify existing tables, create new tables, and define foreignkey/backref relationships',
                'Constructed frontend dashboards and charts to display data',
                'Implemented parameterized testing; populated local databases, mocked M2M request responses and authorization decorators'
            ]
        },
        {
            url: 'lobbyview',
            org: 'MIT Dept Political Science',
            project: 'LobbyView Database',
            title: 'Full Stack Developer',
            // stack: 'Elasticsearch, python, Angular, D3.js package',
            link: 'https://www.lobbyview.org/',
            span: '2017-2018',
            details: [
                'Spearheaded the design of a new website with new query filters',
                'Overhauled dependency management, Gulp build process, and Angular component architecture',
                'Upgraded ElasticSearch version; Rebuilt custom query functions',
                'Created new word-parsing n-gram cloud visualization & backend'
            ]
        },
        {
            url: 'napa',
            org: 'New American Public Art',
            project: 'Color Commons',
            title: 'Open Sourceror',
            // stack: 'Tornado (python), HTML, D3.js package',
            link: 'http://www.newamericanpublicart.com/color-commons-2017',
            span: '2017-2017',
            details: [
                'Built and managed RESTful API architecture for a public art installation on the Boston Greenway',
                'Developed filtered visualization of frequency of user SMS requests using D3 sunburst visualization'
            ]
        },
        {
            url: 'crafts',
            org: 'Tufts University Crafts Center',
            project: 'Glass/Electronics Dept',
            title: 'Volunteer Coordinator',
            link: 'https://tufts.makernetwork.org/spaces/crafts-center',
            span: '2014-2018',
            details: [
                'Aided students and community members at a public arts & crafts resource center'
            ]
        }
    ],
    projects: [
        {
            url: 'self-nn',
            org: 'Self-Directed',
            project: 'Neural Network Digit Predictor',
            title: '',
            link: 'https://tfjs-mnist-predictor.herokuapp.com/',
            span: '2019',
            details: [
                'Built a web app to support an interactive tensorflow-framework neural network classifier',
                'Used a trained model against canvas input to predict digits drawn by the user'
            ]
        },
        {
            url: 'ml-nn',
            org: 'Intro to Machine Learning',
            project: 'Neural Network',
            title: '',
            link: 'https://github.com/syd-strzempko/archived-projects/tree/master/neural-network',
            span: '2018',
            details: [
                'Built a neural network from scratch to run with multi-depth/width settings on nonbinary features',
                'Graphed test set classifier accuracy to portray best depth and width combinations for learning'
            ]
        },
        {
            url: 'um',
            org: 'Machine Structure & Assembly Programming',
            project: 'Universal Machine',
            title: '',
            link: 'https://github.com/syd-strzempko/archived-projects/tree/master/universal-machine',
            span: '2018',
            details: [
                'Coded a fully functional (13-command) UM with segmented memory in C as part of a paired team',
                'Tested refactored code against progressively better benchmarks with GDB/DDB debugging'
            ]
        },
        {
            url: 'ml-classify',
            org: 'Intro to Machine Learning',
            project: 'IBK v J48 Classification',
            title: '',
            link: 'https://github.com/syd-strzempko/archived-projects/tree/master/IBK-v-J48',
            span: '2018',
            details: [
                'Demonstrated iterable kNN and decision tree learning on multi-feature classification sets',
                'Graphed best test set accuracy as a product of variation in features vs train set size'
            ]
        }
    ]
  },
  reducers: {},
});

// List selectors

export const selectJobs = (state) => {
    return state.work.jobs;
};

export const selectProjects = (state) => {
    return state.work.projects;
};

export const selectCombined = (state) => {
    return [ ...state.work.jobs, ...state.work.projects ];
};

// Element selectors

export const selectJob = (state, url) => {
    return state.work.jobs.find((job) => job.url === url);
}

export const selectProject = (state, url) => {
    return state.work.projects.find((project) => project.url === url);
}

export const selectSingle = (state, url) => {
    const combined = selectCombined(state);
    return combined.find((single) => single.url === url);
}

export default slice.reducer;
