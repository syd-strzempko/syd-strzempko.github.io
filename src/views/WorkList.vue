<template>
  <div class="content worklist">
    <Header title="Work History" subtitle="& Selected Projects" />
    <div class="wrapper">
      <WorkTile v-for="project in combinedWorkAndProjects" v-bind:key="project.work.id" v-bind:job="project.work" v-bind:isjob="project.isJob" />
    </div>
    <ResumeDL />
  </div>
</template>

<script>
import Header from '@/components/Header.vue'
import ResumeDL from '@/components/ResumeDL.vue'
import WorkTile from '@/components/WorkTile.vue'

export default {
  props: {
    data: Object 
  },
  name: 'WorkList',
  components: {
    Header,
    ResumeDL,
    WorkTile
  },
  computed: {
    combinedWorkAndProjects() {
      return this.data.jobs.map(function (x) { return {work: x, isJob: true} })
        .concat(this.data.projects.map(function (x) { return {work: x, isJob: false} }));
    }
  }
}
</script>

<style scoped>
.wrapper {
  display: flex;
  flex-flow: column nowrap;
}
</style>
