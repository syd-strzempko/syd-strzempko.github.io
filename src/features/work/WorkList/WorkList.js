import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { selectWorkComplete } from '../workSlice';
import styles from '../work.module.css';
import Layout from '../../../components/Layout';
import ProgressiveImage from '../../../components/ProgressiveImage';

export const WorkList = () => {
  const jobs = useSelector(selectWorkComplete);

  return (
    <Layout>
      <div className={styles.wrapper}>
        {jobs?.map((job, i) => {
          return (
            <Link to={`/work/${job.url}`} key={i}>
              <div className={styles.tile}>
                <ProgressiveImage image={job.images[0]} tile />
                <div className={styles.tileTitle}>{job.title ? job.org : job.project}</div>
              </div>
            </Link>
          );
        })}
      </div>
    </Layout>
  );
}