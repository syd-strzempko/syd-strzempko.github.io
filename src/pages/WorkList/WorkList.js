import { Fragment } from 'react';
import Layout from '../../components/Layout';
import NavBar from '../../components/NavBar';
import './WorkList.css';

const WorkList = () => {
  return (
    <Layout>
      <NavBar />
      <Fragment>WorkIndex</Fragment>
    </Layout>
  );
}

export default WorkList;