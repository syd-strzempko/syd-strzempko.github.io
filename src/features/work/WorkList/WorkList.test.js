import { WorkList } from './WorkList';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import * as reactRedux from 'react-redux';
import { mockStore } from '../workSlice.test';

jest.mock('../../../components/Circles/Circles');

// Note: MemoryRouter allows for nested Link components to exist inside of & access stack of mocked router
describe('WorkList', () => {
    // Mock useSelectors
    beforeEach(() => { useSelectorMock.mockImplementation(selector => selector(mockStore)); });
    afterEach(() => { useSelectorMock.mockClear(); });

    const useSelectorMock = reactRedux.useSelector;

    // Unit tests
    test('renders correctly', () => {
        let view = renderer.create(
            <MemoryRouter>
                <WorkList />
            </MemoryRouter>
        ).toJSON();
        expect(view).toMatchSnapshot();
    });
});