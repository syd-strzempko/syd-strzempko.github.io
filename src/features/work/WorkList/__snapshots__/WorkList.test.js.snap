// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkList renders correctly 1`] = `
Array [
  <mock-Circles />,
  <div
    className="layout"
  >
    <div
      className="navWrapper"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/work"
        onClick={[Function]}
      >
        Work
      </a>
      <a
        href="/art"
        onClick={[Function]}
      >
        Art
      </a>
    </div>
    <div
      className="navClose"
      onClick={[Function]}
    >
      X
    </div>
    <div
      className="navOpen visible"
      onClick={[Function]}
    >
      <img
        alt="menu"
        src="hamburger.png"
      />
    </div>
    <div
      className="wrapper"
    >
      <a
        href="/work/job_test"
        onClick={[Function]}
      >
        <div
          className="tile"
        >
          <img
            alt="work 0"
            className="tile"
            src="mock_long.jpg"
            style={Object {}}
          />
          <div
            className="tileTitle"
          >
            test
          </div>
        </div>
      </a>
      <a
        href="/work/project_test"
        onClick={[Function]}
      >
        <div
          className="tile"
        >
          <img
            alt="work 1"
            className="tile"
            src="mock_long.jpg"
            style={Object {}}
          />
          <div
            className="tileTitle"
          >
            test
          </div>
        </div>
      </a>
    </div>
  </div>,
]
`;
